internal/controllers/machinehealthcheck/machinehealthcheck_targets_test.go

```diff
@@ -426,6 +449,14 @@ func TestHealthCheckTargets(t *testing.T) {
 			expectedNeedsRemediationCondition: []clusterv1.Condition{machineFailureMsgCondition},
 			expectedNextCheckTimes:            []time.Duration{},
 		},
+		{
+			desc:                              "health check with empty unhealthy conditions",
+			targets:                           []healthCheckTarget{nodeGoneAwayEmptyConditions},
+			expectedHealthy:                   []healthCheckTarget{},
+			expectedNeedsRemediation:          []healthCheckTarget{nodeGoneAwayEmptyConditions},
+			expectedNeedsRemediationCondition: []clusterv1.Condition{nodeGoneAwayCondition},
+			expectedNextCheckTimes:            []time.Duration{},
+		},
```

Comment:
Should we add a test where that ensures that conditions are ignored (a node with Ready=false is considered healthy if unhealthy conditions are not defined in MHC)

Reply:
So you mean we need to check against a node like this?

```go
Node:        &corev1.Node{
	Status: corev1.NodeStatus{
		Conditions: []corev1.NodeCondition{
			corev1.NodeCondition{
				Type: corev1.NodeReady,
				Status: corev1.ConditionFalse,
			},
		},
	},
},
}
```

Reply:
Yes and no. I think we need one test case with Ready=False where the Machine/Node won't be remediated (the current one is remediated because of node missing)